### ISSUE TITLE
Pull content only succeeds first time after connection created

### ISSUE THREAD

**Issue by westminsterfl** [10up/distributor#1025]
I have been stumped by a peculiar problem: I was able to successfully configure a new External Connection and pull content, but it fails upon subsequent attempts -- only the first pull works.

After that, I get the following error:

`Could not pull content from connection due to error.
Could not connect to API endpoint.`

What's curious is that it succeeded in pulling 3 posts I selected and then failed upon subsequent attempts. I deleted the connection and re-created it, and the exact same thing happened again.

I verified that the two sites both have the same version of the plugin installed...

Originating site:
Wordpress 6.1.1
Distributor 1.9.0

Pulling site:
Wordpress 6.1.1
Distributor 1.9.0

Any advice you all have in troubleshooting would be greatly appreciated. I can't keep re-creating the connection every time I want to pull in content!

**Comment by jeffpaul** [comment]
@westminsterfl can you share a site health export from the originating site for us to see if there's something there that might be inadvertently breaking the REST API connection?

**Comment by westminsterfl** [comment]
Yes, I can do that. Please see below. Unfortunately it is for the entire multisite network, but that'll have to do, right?

`
### wp-core ###

version: 6.1.1
site_language: en_US
user_language: en_US
timezone: America/New_York
permalink: /news/%category%/%year%/%monthnum%/%postname%/
https_status: true
multisite: true
user_registration: false
blog_public: 1
default_comment_status: open
environment_type: production
site_count: 12
network_count: 1
user_count: 6
dotorg_communication: true

### wp-dropins (3) ###

advanced-cache.php: true
maintenance.php: true
object-cache.php: true

### wp-active-theme ###

name: wcf-wordpress (wcf-wordpress)
version: 1.0.0
author: Underscores.me
author_website: http://underscores.me/
parent_theme: none
theme_features: core-block-patterns, widgets-block-editor, automatic-feed-links, title-tag, post-thumbnails, menus, html5, custom-background, customize-selective-refresh-widgets, custom-logo, custom-header, infinite-scroll, jetpack-responsive-videos, jetpack-content-options, widgets
theme_path: /nas/content/live/westminsterfl/wp-content/themes/wcf-wordpress
auto_update: Enabled

### wp-themes-inactive (5) ###

Hello Elementor: version: 2.6.1, author: Elementor Team, Auto-updates enabled
Twenty Nineteen: version: 2.4, author: the WordPress team, Auto-updates enabled
Twenty Twenty: version: 2.1, author: the WordPress team, Auto-updates enabled
Twenty Twenty-One: version: 1.7, author: the WordPress team, Auto-updates enabled
Twenty Twenty-Two: version: 1.3, author: the WordPress team, Auto-updates enabled

### wp-mu-plugins (5) ###

Force Strong Passwords - WPE Edition: version: 1.8.0, author: Jason Cosper
WP Engine Cache Plugin: version: 1.1.2, author: WP Engine
WP Engine Seamless Login Plugin: version: 1.6.0, author: WP Engine
WP Engine Security Auditor: version: 1.0.10, author: wpengine
WP Engine System: version: 5.0.1, author: WP Engine

### wp-plugins-active (28) ###

301 Redirects Pro: version: 6.04, author: WebFactory Ltd, Auto-updates disabled
ActiveDEMAND: version: 0.2.35, author: JumpDEMAND Inc., Auto-updates enabled
Advanced Accordion Block: version: 4.6.0, author: spider-themes, Auto-updates disabled
Advanced Custom Fields: version: 6.0.7, author: WP Engine, Auto-updates enabled
Advanced Custom Fields: Extended: version: 0.8.9.2, author: ACF Extended, Auto-updates enabled
Advanced Custom Fields PRO: version: 6.0.7, author: WP Engine, Auto-updates enabled
Advanced Sidebar Menu: version: 9.0.5, author: OnPoint Plugins, Auto-updates enabled
Akismet Anti-Spam: version: 5.0.2, author: Automattic, Auto-updates enabled
Better Search Replace: version: 1.4.2, author: WP Engine, Auto-updates disabled
Bulk Page Creator: version: 1.1.4, author: Dagan Lev, Auto-updates enabled
Classic Widgets: version: 0.3, author: WordPress Contributors, Auto-updates enabled
Cloudflare: version: 4.11.0, author: Cloudflare, Inc., Auto-updates enabled
Coming Soon Page, Maintenance Mode, Landing Pages & WordPress Website Builder by SeedProd: version: 6.15.7, author: SeedProd, Auto-updates enabled
Custom Post Type UI: version: 1.13.4, author: WebDevStudios, Auto-updates enabled
Distributor: version: 1.9.0, author: 10up Inc., Auto-updates enabled
Easy FancyBox: version: 1.9.5, author: RavanH, Auto-updates enabled
Events Manager: version: 6.3, author: Pixelite, Auto-updates enabled
FileBird Lite: version: 5.1.4, author: Ninja Team, Auto-updates enabled
Hotjar: version: 1.0.14, author: Hotjar, Auto-updates disabled
Jetpack: version: 11.9, author: Automattic, Auto-updates enabled
Lightbox for Gallery & Image Block: version: 1.13, author: Johannes Kinast <johannes@travel-dealz.de>, Auto-updates enabled
Nested Pages: version: 3.2.3, author: Kyle Phillips, Auto-updates enabled
Redirection: version: 5.3.9, author: John Godley, Auto-updates enabled
WordPress Importer: version: 0.8, author: wordpressdotorg, Auto-updates disabled
WPCode Pro: version: 2.0.8.1, author: WPCode, Auto-updates disabled
WP MAPS PRO: version: 5.4.3, author: flippercode, Auto-updates enabled
WP STAGING PRO: version: 4.1.8, author: WP-STAGING (latest version: 4.6.1), Auto-updates enabled
Yoast SEO: version: 20.2.1, author: Team Yoast, Auto-updates enabled

### wp-plugins-inactive (12) ###

301 Redirects: version: 2.72, author: WebFactory Ltd, Auto-updates disabled
Agile Store Locator: version: 4.6.36, author: AGILELOGIX, Auto-updates enabled
Contact Form 7: version: 5.7.4, author: Takayuki Miyoshi, Auto-updates enabled
Custom Permalinks: version: 2.5.2, author: Sami Ahmed Siddiqui, Auto-updates disabled
Elementor: version: 3.11.3, author: Elementor.com, Auto-updates enabled
Network Media Library: version: 1.5.0, author: John Blackbourn, Dominik Schilling, Frank Bültge, Auto-updates enabled
Permalinks Customizer: version: 2.8.2, author: YAS Global Team, Auto-updates disabled
WP All Export - ACF Export Add-On Pro: version: 1.0.4, author: Soflyy, Auto-updates disabled
WP All Export Pro: version: 1.8.1, author: Soflyy, Auto-updates disabled
WP All Import - ACF Add-On: version: 3.3.6, author: Soflyy, Auto-updates enabled
WP All Import Pro: version: 4.7.1, author: Soflyy, Auto-updates enabled
WPCode Lite: version: 2.0.8.1, author: WPCode, Auto-updates enabled

### wp-media ###

image_editor: WP_Image_Editor_Imagick
imagick_module_version: 1691
imagemagick_version: ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25 https://imagemagick.org
imagick_version: 3.6.0
file_uploads: File uploads is turned off
post_max_size: 100M
upload_max_filesize: 50M
max_effective_size: 50 MB
max_file_uploads: 20
imagick_limits: 
	imagick::RESOURCETYPE_AREA: 137 MB
	imagick::RESOURCETYPE_DISK: 1073741824
	imagick::RESOURCETYPE_FILE: 786432
	imagick::RESOURCETYPE_MAP: 512 MB
	imagick::RESOURCETYPE_MEMORY: 256 MB
	imagick::RESOURCETYPE_THREAD: 1
imagemagick_file_formats: 3FR, 3G2, 3GP, AAI, AI, APNG, ART, ARW, AVI, AVIF, AVS, BGR, BGRA, BGRO, BIE, BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CR2, CR3, CRW, CUR, CUT, DATA, DCM, DCR, DCX, DDS, DFONT, DJVU, DNG, DOT, DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2, EPT3, ERF, EXR, FAX, FILE, FITS, FRACTAL, FTP, FTS, G3, G4, GIF, GIF87, GRADIENT, GRAY, GRAYA, GROUP4, GV, H, HALD, HDR, HEIC, HISTOGRAM, HRZ, HTM, HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL, ISOBRL6, J2C, J2K, JBG, JBIG, JNG, JNX, JP2, JPC, JPE, JPEG, JPG, JPM, JPS, JPT, JSON, K25, KDC, LABEL, M2V, M4V, MAC, MAGICK, MAP, MASK, MAT, MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPG, MRW, MSL, MSVG, MTV, MVG, NEF, NRW, NULL, ORF, OTB, OTF, PAL, PALM, PAM, PANGO, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PGX, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64, PNG8, PNM, POCKETMOD, PPM, PREVIEW, PS, PS2, PS3, PSB, PSD, PTIF, PWP, RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGBA, RGBO, RGF, RLA, RLE, RMF, RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF, STEGANO, SUN, SVG, SVGZ, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TTC, TTF, TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIDEO, VIFF, VIPS, VST, WBMP, WEBM, WEBP, WMF, WMV, WMZ, WPG, X, X3F, XBM, XC, XCF, XPM, XPS, XV, XWD, YCbCr, YCbCrA, YUV
gd_version: 2.3.0
gd_formats: GIF, JPEG, PNG, WebP, BMP, XPM
ghostscript_version: unknown

### wp-server ###

server_architecture: Linux 5.10.147+ x86_64
httpd_software: nginx
php_version: 7.4.33.1 64bit
php_sapi: fpm-fcgi
max_input_variables: 10000
time_limit: 43300
memory_limit: 512M
max_input_time: -1
upload_max_filesize: 50M
php_post_max_size: 100M
curl_version: 7.81.0 OpenSSL/3.0.2
suhosin: false
imagick_availability: true
pretty_permalinks: true
htaccess_extra_rules: true

### wp-database ###

extension: mysqli
server_version: 5.7.40-43-log
client_version: mysqlnd 7.4.33.1
max_allowed_packet: 16777216
max_connections: 500

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /nas/content/live/westminsterfl/wp-content
WP_PLUGIN_DIR: /nas/content/live/westminsterfl/wp-content/plugins
WP_MEMORY_LIMIT: 64M
WP_MAX_MEMORY_LIMIT: 512M
WP_DEBUG: false
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: true
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: Undefined
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable
mu-plugins: writable

### jetpack ###

site_id: 198272007
ssl_cert: No
time_diff: undefined
version_option: 11.9:1678249128
old_version: 11.8.4:1676521260
public: Public
master_user: #1 wmeltzer_c9fvb
current_user: #1 wmeltzer_c9fvb
tokens_set: Blog User
blog_token: jQRy^@&Tb2^ncjzj4@D5ybLu2WKn522U
user_token: v1hXw^K@3Z%iuBmilR87sv@DnhkVtqZg
version: 11.9
jp_plugin_dir: /nas/content/live/westminsterfl/wp-content/plugins/jetpack/
plan: free
protect_header: {"trusted_header":"HTTP_CF_CONNECTING_IP","segments":0,"reverse":false}
full_sync: {"started":"Sat, 02 Oct 2021 17:49:17 +0000","finished":"Sat, 02 Oct 2021 17:49:25 +0000","progress":{"options":{"finished":true},"functions":{"finished":true},"constants":{"finished":true},"users":{"total":"1","sent":1,"finished":true,"last_sent":"1"},"network_options":{"finished":true}},"config":{"options":true,"functions":true,"constants":true,"users":[1],"network_options":true}}
sync_size: undefined
sync_lag: 0 seconds
full_sync_size: undefined
full_sync_lag: 0 seconds
idc_urls: {"home":"https:\/\/westminstercommunitiesfl.org","siteurl":"https:\/\/westminstercommunitiesfl.org","WP_HOME":"","WP_SITEURL":""}
idc_error_option: false
idc_optin: false
cxn_tests: All Pass.

### distributor ###

0: 1.9.0
1: 
	Override Author Byline: Yes
	Media Handling: Featured image only
2: 
	Westminster Manor: 
- Blog ID: 2
- URL: https://www.westminstermanorfl.org
- Registered: 2021-10-08 22:53:31
- Last updated: 2023-02-02 11:54:03
	Westminster Oaks: 
- Blog ID: 3
- URL: https://www.westminsteroaksfl.org
- Registered: 2021-12-13 15:02:16
- Last updated: 2023-02-02 11:51:32
	Westminster Point Pleasant: 
- Blog ID: 4
- URL: https://www.westminsterpointpleasantfl.org
- Registered: 2021-12-15 19:28:15
- Last updated: 2023-03-02 16:04:25
	Westminster Palms: 
- Blog ID: 5
- URL: https://www.westminsterpalmsfl.org
- Registered: 2021-12-18 19:59:35
- Last updated: 2023-02-02 11:35:38
	Westminster Shores: 
- Blog ID: 6
- URL: https://www.westminstershoresfl.org
- Registered: 2021-12-18 21:52:49
- Last updated: 2023-03-07 14:45:27
	Westminster Suncoast: 
- Blog ID: 7
- URL: https://www.westminstersuncoastfl.org
- Registered: 2021-12-24 17:11:28
- Last updated: 2023-03-07 14:58:37
	Westminster St. Augustine: 
- Blog ID: 8
- URL: https://www.westminsterstaugustinefl.org
- Registered: 2021-12-24 19:23:10
- Last updated: 2023-02-08 13:35:38
	Westminster Towers: 
- Blog ID: 9
- URL: https://www.westminstertowersfl.org
- Registered: 2021-12-26 16:30:08
- Last updated: 2023-03-02 15:57:33
	Westminster Woods on Julington Creek: 
- Blog ID: 10
- URL: https://www.westminsterwoodsfl.org
- Registered: 2021-12-26 18:48:39
- Last updated: 2023-02-01 13:36:39
	Westminster Winter Park: 
- Blog ID: 11
- URL: https://www.westminsterwinterparkfl.org
- Registered: 2021-12-26 19:58:39
- Last updated: 2023-02-01 14:06:37
	Westminster Baldwin Park: 
- Blog ID: 12
- URL: https://www.westminsterbaldwinpark.com
- Registered: 2021-12-26 20:53:58
- Last updated: 2023-02-02 11:56:49
3: N/A

`

**Comment by jeffpaul** [comment]
@westminsterfl can you try updating to Distributor 2.0.0 and either using the automated connection process or utilize a directly created Application Password from your connected site (different from an actual user password)?

If that still fails, could you provide a screenshot of the External Connection (feel free to obscure the username and password).

**Comment by westminsterfl** [comment]
@jeffpaul I actually attempted to upgrade to 2.0.0 on both sites (the origin and the destination), but there appears to be a conflict with upgrading the plugin. It causes the entire site to fail with PHP error:

>> Uncaught Error: Class '\Puc_v4p13_Vcs_PluginUpdateChecker' not found in /nas/content/live/[...]/wp-content/plugins/distributor-stable/vendor/yahnis-elsts/plugin-update-checker/Puc/v4p13/Factory.php:138 Stack trace: #0 /nas/content/live/[...]/wp-content/plugins/distributor-stable/distributor.php(175): Puc_v4p13_Factory::buildUpdateChecker('https://github....', '/nas/content/li...', 'distributor') #1 /nas/content/live/[...]/wp-settings.php(407): include_once('/nas/content/li...') #2 /nas/content/live/[...]/wp-config.php(133): require_once('/nas/content/li...') #3 /nas/content/live/[...]/wp-load.php(50): require_once('/nas/content/li...') #4 /nas/content/live/[...]/wp-admin/admin-ajax.php(22): require_once('/nas/content/li...') #5 {main} thrown on line 138 in file wp-content/plugins/distributor-stable/vendor/yahnis-elsts/plugin-update-checker/Puc/v4p13/Factory.php

I have a development environment I can test this upgrade on by uninstalling and reinstalling, but as that may wipe out my configuration, I'd really rather not.

**Comment by jeffpaul** [comment]
@peterwilsoncc see the above details on a PHP error when updating to Distributor v2 on a multisite network ^

**Comment by peterwilsoncc** [comment]
@westminsterfl  I've attempted to reproduce the updating error in the comment above without success.

Are you able to see if any of the other premium plugins are also using the same updating system, first by searching for `Puc_v4p13_Vcs_PluginUpdateChecker` and if that doesn't appear `Vcs_PluginUpdateChecker`.

Also, are you seeing the error when running updates manually in the admin or when running auto-updates?

Thank you1
